reconcile/vault_replication.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import re
+
+from reconcile.utils.vault import VaultClient, SecretNotFound
+from reconcile.gql_definitions.jenkins_configs import jenkins_configs
+
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.gql_definitions.vault_instances import vault_instances
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultReplicationV1,
+)
+from reconcile.gql_definitions.vault_policies.vault_policies import (
+    VaultPolicyV1,
+)
+
+from reconcile.utils import gql
+
+from typing import List
+
+QONTRACT_INTEGRATION = "vault-replication"
+
+
+def copy_vault_secret(
+    dry_run: bool, source_vault: VaultClient, dest_vault: VaultClient, path: str
+) -> None:
+
+    secret_dict = {"path": path, "version": "LATEST"}
+    secrets, version = source_vault.read_all_version(secret_dict)
+    for secret in secrets:
```

Comment:
it might be well true that my mental model of vault secrets is wrong, so please please let me know if that is the case :)

i thought the variable `secrets` contains all the `keys` that make up a single secret. if that is true (if not please let me know), updating the destination vault with `dest_vault.write` should be a single operation and not done within the loop for each key within the secret?

Reply:
100% True, I misunderstood the implementation of the vault client that we have in qontract-reconcile, changed to match the expected behaviour.